src/parser/parser_visitor.py

```diff
@@ -177,16 +184,26 @@ def visitCreateDefinitions(
     def visitColumnDeclaration(
             self, ctx: evaql_parser.ColumnDeclarationContext):
 
-        data_type, dimensions = self.visit(ctx.columnDefinition())
+        data_type, dimensions, unique = self.visit(ctx.columnDefinition())
         column_name = self.visit(ctx.uid())
 
         if column_name is not None:
-            return ColumnDefinition(column_name, data_type, dimensions)
+            return ColumnDefinition(column_name, data_type, dimensions, unique)
 
     def visitColumnDefinition(self, ctx: evaql_parser.ColumnDefinitionContext):
 
         data_type, dimensions = self.visit(ctx.dataType())
-        return data_type, dimensions
+
+        constraint_count = len(ctx.columnConstraint())
+        unique = False
```

Comment:
Rather than checking whether it is unique or not, we should return the constraint type.